src/object/srv_obj.c

```diff
@@ -2304,6 +2305,19 @@ obj_inflight_io_check(struct ds_cont_child *child, uint32_t opc, uint32_t flags)
 	if (!obj_is_modification_opc(opc))
 		return 0;
 
+	if (child->sc_pool->spc_pool->sp_rebuilding) {
+		uint32_t version;
+
+		ds_rebuild_running_query(child->sc_pool_uuid, RB_OP_REBUILD,
+					 &version, NULL, NULL);
+		if (version != 0 && version < rpc_map_ver) {
```

Comment:
because with delay rebuild patch, the pool map can be changed during other pool rebuilding, so if there are new extending/reintegrate happened, the pool map might change, and broadcast to the client, then client use the new pool map to generate the layout, but if the extend/reint/rebuild are not started on the server yet. Then it may corrupt the data.

So we use this check to let the inflight I/O wait until the correspondent rebuild start.